go/integTest/glide_test_suite_test.go

```diff
@@ -391,3 +398,247 @@ func (suite *GlideTestSuite) GenerateLargeUuid() string {
+		assert.Fail(suite.T(), "Unsupported client type")
+		return nil
+	}
+}
+
+func (suite *GlideTestSuite) createStandaloneClientWithSubscriptions(
+	config *api.StandaloneSubscriptionConfig,
+) api.GlideClientCommands {
+	clientConfig := suite.defaultClientConfig().WithSubscriptionConfig(config)
+	return suite.client(clientConfig)
+}
+
+func (suite *GlideTestSuite) createClusterClientWithSubscriptions(
+	config *api.ClusterSubscriptionConfig,
+) api.GlideClusterClientCommands {
+	clientConfig := suite.defaultClusterClientConfig().WithSubscriptionConfig(config)
+	return suite.clusterClient(clientConfig)
+}
+
+type TestChannelMode int
+
+const (
+	ExactMode TestChannelMode = iota
+	PatternMode
+)
+
+type ChannelDefn struct {
+	Mode    TestChannelMode
+	Channel string
+}
+
+var callbackCtx = sync.Map{}
+
+const MESSAGE_DELIVERY_DELAY = 500 // ms
+
+type MessageReadMethod int
+
+const (
+	CallbackMethod MessageReadMethod = iota
+	WaitForMessageMethod
+	SignalChannelMethod
+	SyncLoopMethod
+)
+
+// verifyPubsubMessages verifies that subscribers received the expected messages
+// For single subscriber, pass a map with a single queue using any key (e.g., 1)
+// For pattern subscriptions, the expectedMessages keys should be the pattern values
+func (suite *GlideTestSuite) verifyPubsubMessages(
+	t *testing.T,
+	expectedMessages map[string]string,
+	queues map[int]*api.PubSubMessageQueue,
+	messageReadMethod MessageReadMethod,
+) {
+	switch messageReadMethod {
+	case CallbackMethod:
+		receivedMessages := make(map[string]map[string]string)
+		callbackCtx.Range(func(key, value any) bool {
+			keyStr := key.(string)
+			parts := strings.Split(keyStr, "-")
+			clientId := parts[0]
+			message := value.(*api.PubSubMessage)
+			if _, exists := receivedMessages[clientId]; !exists {
+				receivedMessages[clientId] = make(map[string]string)
+			}
+			// For pattern subscriptions, use the pattern value as the key
+			// For channel subscriptions, use the channel name as the key
+			messageKey := message.Channel
+			if !message.Pattern.IsNil() {
+				messageKey = message.Pattern.Value()
+			}
+			receivedMessages[clientId][messageKey] = message.Message
+			return true
+		})
+
+		// Verify each subscriber received the expected messages
+		for clientId, messages := range receivedMessages {
+			if !assert.Equal(t, expectedMessages, messages, "Messages mismatch for client %s", clientId) {
+				t.FailNow()
+			}
+		}
+
+	case WaitForMessageMethod:
+		for clientId, queue := range queues {
+			receivedMessages := make(map[string]string)
+			for expectedKey := range expectedMessages {
+				select {
+				case msg := <-queue.WaitForMessage():
+					// For pattern subscriptions, use the pattern value as the key
+					// For channel subscriptions, use the channel name as the key
+					messageKey := msg.Channel
+					if !msg.Pattern.IsNil() {
+						messageKey = msg.Pattern.Value()
+					}
+					receivedMessages[messageKey] = msg.Message
+				case <-time.After(5 * time.Second):
```

Comment:
Please move waiting timeout to constants above

Reply:
Done